crates/task-impls/src/da.rs

```diff
@@ -260,8 +263,15 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 let encoded_transactions_hash = Sha256::digest(&encoded_transactions);
 
                 // sign the encoded transactions as opposed to the VID commitment
-                let signature =
-                    TYPES::SignatureKey::sign(&self.private_key, &encoded_transactions_hash);
+                let Ok(signature) =
+                    TYPES::SignatureKey::sign(&self.private_key, &encoded_transactions_hash)
+                else {
+                    // TODO is this correct?
+                    // Should we be doing more?
+                    error!("Failed to sign block payload!");
+                    return None;
+                };
```

Comment:
Looks fine to me. I think it's hard to fail in signing sth, unless it's a very extreme corner case.